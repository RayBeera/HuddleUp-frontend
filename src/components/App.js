import { BrowserRouter, Navigate, Route, Routes } from "react-router-dom";
import { useAuthContext } from "../hooks/useAuthContext";

import SignUp from "./SignUp";
import Login from "./Login";
import Home from "./Home";
import Navbar from "./Navbar";
import MyProfile from "./MyProfile";
import JoinEvents from "./JoinEvents";
import MyEvents from "./MyEvents";
import LogOut from "./Logout";
import CreateEvent from "./CreateEvent";

import "../styles/app.css";

function App() {
  const { user } = useAuthContext();

  return (
    <div className="App">
      <div>
        <BrowserRouter>
          <Navbar />
          <Routes>
            <Route path="/" element={<Home />}></Route>
            <Route path="/signup" element={<SignUp />}></Route>
            <Route path="/login" element={<Login />}></Route>
            <Route path="/myprofile" element={<MyProfile />}></Route>
            <Route path="/createevent" element={<CreateEvent />}></Route>
            {/* <Route
              path="/myProfile"
              element={user ? <MyProfile /> : <Navigate to="/login" />}
            /> */}
            <Route
              path="/joinevents"
              element={user ? <JoinEvents /> : <Navigate to="/login" />}
            />
            <Route
              path="/myevents"
              element={user ? <MyEvents /> : <Navigate to="/login" />}
            />
          </Routes>
        </BrowserRouter>
      </div>
    </div>
  );
}

export default App;
