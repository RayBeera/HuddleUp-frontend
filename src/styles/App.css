@import url("https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,300;0,400;0,500;0,600;0,700;0,800;1,200;1,300;1,400;1,500;1,600;1,700;1,800&display=swap");
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: "poppins";
  text-align: center;
 
}
/* .backgroundLogo{
  background-image: url(../../public/AdobeStock_375037420.jpeg);
  background-size: cover;
  opacity: 0.6;
  align-items: center;
  display: grid;
  
} */