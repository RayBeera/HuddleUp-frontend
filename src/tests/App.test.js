import React from "react";
import { render } from "@testing-library/react";
import { AuthContextProvider } from "../contexts/AuthContext";
import App from "../components/App";

describe("App", () => {
  it("renders App correctly", () => {
    const { asFragment } = render(
      <AuthContextProvider>
        <App />
      </AuthContextProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
