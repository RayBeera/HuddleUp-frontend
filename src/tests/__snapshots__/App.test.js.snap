// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="navbar"
      >
        <div
          class="rightSide"
        >
          <div
            class="links"
            id=""
          >
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/login"
            >
              Login
            </a>
            <a
              href="/signup"
            >
              Sign up
            </a>
          </div>
          <div
            class="nav-button"
          >
            <svg
              color="white"
              fill="currentColor"
              height="2.5rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 24 24"
              width="2.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 15h18v-2H3v2zm0 4h18v-2H3v2zm0-8h18V9H3v2zm0-6v2h18V5H3z"
              />
            </svg>
          </div>
        </div>
        <div
          class="leftSide"
        />
      </div>
      <div
        class="homeBody"
      >
        <h1
          class="title"
        >
          Huddle Up
        </h1>
        <div
          class="homeText"
        >
          <h3
            style="font-weight: 200;"
          >
            Join us and plan your events and activities!
          </h3>
        </div>
        <div
          class="btns logged-out"
        >
          <button
            class="loginBtn"
          >
            Login
          </button>
          <button
            class="signUpBtn"
          >
            Sign up
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
